src/Columns/ColumnUnique.h

```diff
@@ -298,6 +300,51 @@ size_t ColumnUnique<ColumnType>::getNullValueIndex() const
     return 0;
 }
 
+template <typename ColumnType>
+void ColumnUnique<ColumnType>::insertWithGetTransIndex(const IColumn & src, size_t start, size_t length,
+                                                        std::unordered_map<UInt64, UInt64>& trans)
+{
+    const ColumnType * src_column;
+    const NullMap * null_map = nullptr;
+    if (auto * nullable_column = checkAndGetColumn<ColumnNullable>(src))
+    {
+        src_column = typeid_cast<const ColumnType *>(&nullable_column->getNestedColumn());
+        null_map = &nullable_column->getNullMapData();
+    }
+    else
+        src_column = typeid_cast<const ColumnType *>(&src);
+
+    if (src_column == nullptr)
+        throw Exception("Invalid column type for ColumnUnique::insertWithGetTransIndex. Expected " + column_holder->getName() +
+                        ", got " + src.getName(), ErrorCodes::ILLEGAL_COLUMN);
+
+    auto column = getRawColumnPtr();
+    for (size_t row = start; row < length; ++row)
```

Comment:
Suspicious condition. Should not it be like `row < start + length`?

Reply:
Yes, should be. Currently the start always be zero from outside, need fix it.